python/dpu_utils/utils/richpath.py

```diff
@@ -66,43 +99,53 @@ def path(self) -> str:
         To create a remote path, just prefix it appropriately and pass
         in the path to the .json configuration.
         """
         if path.startswith(AZURE_PATH_PREFIX):
-            assert azure_info_path is not None, "An AzurePath cannot be created when azure_info_path is None."
             # Strip off the AZURE_PATH_PREFIX:
             path = path[len(AZURE_PATH_PREFIX):]
             account_name, container_name, path = path.split('/', 2)
 
-            with open(azure_info_path, 'r') as azure_info_file:
-                azure_info = json.load(azure_info_file)
-            account_info = azure_info.get(account_name)
-            if account_info is None:
-                raise Exception("Could not find access information for account '%s'!" % (account_name,))
-
-            sas_token = account_info.get('sas_token')
-            account_key = account_info.get('account_key')
-            if sas_token is not None:
-                assert not sas_token.startswith('?'), 'SAS tokens should not start with "?". Just delete it.'  #  https://github.com/Azure/azure-storage-python/issues/301
-                blob_service = BlockBlobService(account_name=account_name,
-                                                sas_token=sas_token)
-            elif account_key is not None:
-                blob_service = BlockBlobService(account_name=account_name,
-                                                account_key=account_key)
+            if azure_info_path is not None:
+                with open(azure_info_path, 'r') as azure_info_file:
+                    azure_info = json.load(azure_info_file)
+                azure_info = azure_info.get(account_name)
+                if azure_info is None:
+                    raise Exception("Could not find access information for account '%s'!" % (account_name,))
+
+                account_endpoint = azure_info.get('endpoint', "https://%s.blob.core.windows.net/" % account_name)
+                cache_location = azure_info.get('cache_location')
+
+                if azure_info.get('connection_string') is not None:
+                    container_client = ContainerClient.from_connection_string(azure_info.get('connection_string'), container_name)
+                elif azure_info.get('sas_token') is not None:
+                    query_string: str = azure_info.get('sas_token')
+                    if not query_string.startswith('?'):
+                        query_string = '?' + query_string
+                    container_client = ContainerClient.from_container_url(f"{account_endpoint}/{container_name}{query_string}")
+                elif azure_info.get('account_key') is not None:
+                    account_key = azure_info.get('account_key')
+                    connection_string = f"AccountName={account_name};AccountKey={account_key};BlobEndpoint={account_endpoint};"
+                    container_client = ContainerClient.from_connection_string(connection_string, container_name)

```

Comment:
Use local vars instead of repeated `.get`:

```suggestion
               connection_string =  azure_info.get('connection_string')
               sas_token = azure_info.get('sas_token')
               account_key = azure_info.get('account_key')

                if connection_string is not None:
                    container_client = ContainerClient.from_connection_string(connection_string, container_name)
                elif sas_token is not None:
                    query_string: str = sas_token
                    if not query_string.startswith('?'):
                        query_string = '?' + query_string
                    container_client = ContainerClient.from_container_url(f"{account_endpoint}/{container_name}{query_string}")
                elif account_key is not None:
                    connection_string = f"AccountName={account_name};AccountKey={account_key};BlobEndpoint={account_endpoint};"
                    container_client = ContainerClient.from_connection_string(connection_string, container_name)

```